concordium-consensus/src/Concordium/KonsensusV1/TreeState/Implementation.hs

```diff
@@ -796,15 +794,17 @@ finalizeTransactions = mapM_ removeTrans
         -- They can never be part of any other block after this point.
         forM_ (Map.keys nfn) $
             \deadTransaction -> transactionTable . TT.ttHashMap . at' (getHash deadTransaction) .= Nothing
-        -- Update the non-finalized transactions for the sender
+        -- Remove the transaction from the non finalized transactions.
+        -- If there are no non-finalized transactions left then remove the entry
+        -- for the sender in @ttNonFinalizedTransactions@.
         transactionTable
             . TT.ttNonFinalizedTransactions
             . at' sender
-            ?=! ( anft
+            . non TT.emptyANFT
+            .=! ( anft
```

Comment:
**Nitpick:** As already mentioned in my other comment related to the transaction table internals, this should probably be factored out.

Reply:
good idea